enricher/govulncheck/source/govulncheck.go

```diff
@@ -0,0 +1,245 @@
+		findings, err := e.runGovulncheck(ctx, absModDir, goVersion)
+		if err != nil {
+			log.Errorf("govulncheck on %s: %v", modDir, err)
+			continue
+		}
+
+		if len(findings) == 0 {
+			continue
+		}
+
+		e.addSignals(inv, findings)
+	}
+
+	return nil
+}
+
+func (e *Enricher) addSignals(inv *inventory.Inventory, idToFindings map[string][]*internal.Finding) {
+	for _, pv := range inv.PackageVulns {
+		findings, exist := idToFindings[pv.Vulnerability.Id]
+
+		if !exist {
+			if vulnHasImportsField(pv.Vulnerability, pv.Package) {
+				// If there is symbol information, then analysis has been performed, and
+				// code does not import the vulnerable package, so definitely not called
+				pv.ExploitabilitySignals = append(pv.ExploitabilitySignals, &vex.FindingExploitabilitySignal{
+					Plugin:        Name,
+					Justification: vex.VulnerableCodeNotInExecutePath,
+				})
+			}
+
+			// Otherwise, we don't know if the code is reachable or not.
+			continue
+		}
+
+		// For entries with findings, check if the code is reachable or not by whether there is a trace.
+		isReachable := false
+		for _, f := range findings {
+			if len(f.Trace) > 0 && f.Trace[0].Function != "" {
+				isReachable = true
+				break
+			}
+		}
+
+		if !isReachable {
+			pv.ExploitabilitySignals = append(pv.ExploitabilitySignals, &vex.FindingExploitabilitySignal{
+				Plugin:        Name,
+				Justification: vex.VulnerableCodeNotInExecutePath,
+			})
+		}
+	}
+}
+
+func (e *Enricher) runGovulncheck(ctx context.Context, absModDir string, goVersion string) (map[string][]*internal.Finding, error) {
+	args := []string{"-C", absModDir, "-format", "json", "-mode", "source"}
+	if e.offlineVulnDBPath != "" {
+		args = append(args, "-db=file://"+e.offlineVulnDBPath)
+	}
+	cmd := scan.Command(ctx, append(args, "./...")...)
+	var b bytes.Buffer
+	cmd.Stdout = &b
+	cmd.Env = append(os.Environ(), "GOVERSION=go"+goVersion)
+	if err := cmd.Start(); err != nil {
+		return nil, err
+	}
+	if err := cmd.Wait(); err != nil {
+		return nil, err
+	}
+
+	// Group the output of govulncheck based on the OSV ID.
+	h := &osvHandler{
+		idToFindings: map[string][]*internal.Finding{},
+	}
+	if err := handleJSON(bytes.NewReader(b.Bytes()), h); err != nil {
+		return nil, err
+	}
+
+	return h.idToFindings, nil
+}
+
+type osvHandler struct {
+	idToFindings map[string][]*internal.Finding
+}
+
+func (h *osvHandler) Finding(f *internal.Finding) {
+	h.idToFindings[f.OSV] = append(h.idToFindings[f.OSV], f)
+}
+
+func handleJSON(from io.Reader, to *osvHandler) error {
+	dec := json.NewDecoder(from)
+	for dec.More() {
+		msg := internal.Message{}
+		if err := dec.Decode(&msg); err != nil {
+			return err
+		}
+		if msg.Finding != nil {
+			to.Finding(msg.Finding)
+		}
+	}
+
+	return nil
+}
+
+func vulnHasImportsField(vuln *osvschema.Vulnerability, pkg *extractor.Package) bool {
+	for _, affected := range vuln.Affected {
+		if pkg != nil {
+			// TODO: Compare versions to see if this is the correct affected element
```

Comment:
Create an issue for TODOs and link them like `// TODO(#123): `
Seems like it's not too difficult to use the semantic lib for the version comparison here so maybe we can just add it to the PR?

Reply:
Done: https://github.com/google/osv-scalibr/issues/1559

Can't quite implement it yet as it's a bit difficult without the full offline matching logic from osv-scanner in osv-scalibr, and it's quite unlikely to actually affect any real world situations, so leaving it for later.